storage/storage.cpp

```diff
@@ -1145,4 +1145,19 @@ void Storage::DoClickOnDownloadMap(TCountryId const & countryId)
   if (m_downloadMapOnTheMap)
     m_downloadMapOnTheMap(countryId);
 }
+
+void Storage::ForEachInSubtree(TCountryId const & root, TForEachFunction const & toDo) const
```

Comment:
Here && is not a universal reference because TForEachFunction is alias for function, not a template param. It is rvalue.

small code sample:

using TFn = function<void()>;
void Process(TFn && fn) { fn(); }
void Test() {}

int main()
{
    function<void()> t = Test;
    auto l = [](){};
    Process([](){}); // implicit tmp object, rvalue
    Process(Test); // implicit tmp object, rvalue
    Process(l); // implicit tmp object, rvalue
    // Process(t); // compilation error, because t is lvalue
    Process(move(t)); // ok because param is rvalue